utils/data_uri.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 The kubecfg authors
+//
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+package utils
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+)
+
+func ToDataURL(code string) string {
+	return fmt.Sprintf("data:,%s", url.PathEscape(code))
+}
+
+type dataURIImporter struct {
+}
+
+func newDataURIImporter() *dataURIImporter {
+	return &dataURIImporter{}
+}
+
+func (o *dataURIImporter) RoundTrip(req *http.Request) (*http.Response, error) {
+	content, _, err := expandDataURL(req.URL.String())
+	if err != nil {
+		return nil, err
+	}
+
+	return simpleHTTPResponse(req, http.StatusOK, io.NopCloser(strings.NewReader(content))), nil
+}
+
+func expandDataURL(pathURL string) (string, string, error) {
+	content, err := url.PathUnescape(strings.TrimPrefix(pathURL, "data:,"))
```

Comment:
`TrimPrefix` here makes me nervous, because it will be silently ignored, resulting in odd content for things like `data:text/plain,`

I think this function should accept the `url.URL` (since the caller already has a parsed URL), and look at `u.Opaque` (so we've already split off the scheme).  Then a `[encoding, content] := strings.SplitN(url.PathUnescape(u.Opaque), ",", 2)` - I think it's ok to return an error if `encoding != ""`.

Reply:
oh yeah, I completely ignored the mimetype part. I updated also many other places where I just matched on `data:,` ignoring the possibility that the user would provide encodings to data uris.